src/mechanics/cell/SimpleAssember.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include "mechanics/cell/CellInterface.h"
+#include "mechanics/dofs/GlobalDofVector.h"
+#include "mechanics/dofs/GlobalDofMatrixSparse.h"
+
+namespace NuTo
+{
+class SimpleAssembler
+{
+public:
+    SimpleAssembler(const NuTo::DofContainer<int>& numActiveDofs, const NuTo::DofContainer<int>& numDependentDofs)
+        : mNumActiveDofs(numActiveDofs)
+        , mNumDependentDofs(numDependentDofs)
+    {
+    }
+
+    GlobalDofVector BuildGradient(const std::vector<NuTo::CellInterface*>& cells,
+                                  const std::vector<NuTo::DofType*>& dofTypes) const
+    {
+        GlobalDofVector gradient = ProperlyResizedGlobalVector(dofTypes);
+        for (NuTo::CellInterface* cell : cells)
+        {
+            const DofVector<int> numbering = cell->DofNumbering();
+            const DofVector<double> cellGradient = cell->Gradient();
+
+            for (const DofType* dof : dofTypes)
+            {
+                const Eigen::VectorXi& numberingDof = numbering[*dof];
+                const Eigen::VectorXd& cellGradientDof = cellGradient[*dof];
+                const int numActiveDofs = mNumActiveDofs[*dof];
+                for (int i = 0; i < numberingDof.rows(); ++i)
+                {
+                    int globalDofNumber = numberingDof[i];
+                    double globalDofValue = cellGradientDof[i];
+                    if (globalDofNumber < numActiveDofs)
+                        gradient.J[*dof][globalDofNumber] += globalDofValue;
+                    else
+                        gradient.K[*dof][globalDofNumber - numActiveDofs] += globalDofValue;
+                }
+            }
+        }
+        return gradient;
+    }
+
+    GlobalDofMatrixSparse BuildHessian0(const std::vector<NuTo::CellInterface*>& cells,
```

Comment:
`BuildHessian1` and `BuildHessian2` are going to be copy&paste programming, or something else?

Reply:
If we chose to keep this Hessian012 syntax, code duplication can be avoided with something like

~~~cpp
template <typename TFunction>
GlobalDofMatrixSparse BuildMatrix(cells, dofs, TFunction f)
{ ...
    hessian = f(cell);
... }

GlobalDofMatrixSparse BuildHessian0(cells, dofs)
{
    auto f = (CellInterface& cell) {return cell.Hessian0();};
   return BuildMatrix(cells, dofs, f);
}
~~~

(Same pattern can be used to avoid possible code duplication inside the cell.)

Reply:
The more we implement in the "_not_ operator/template concept", the more work is required to change it later. I am aware of that and I want to ensure that I happily adapt all the code to any concept we decide on.
But AFAIK, for now, the plan is to build the most simple running version that is able to solve something like a linear elastic patch test. This introduces various challenges (dof numbering, node selection, mesh class, ConvertToCells(), ... ). So for now, there is only Gradient() and Hessian() and imo no need to make it more complicated that that.

My code above is just _one_ way to assemble an arbitrary cell method.